plugins/language_haxe.lua

```diff
@@ -0,0 +1,111 @@
+-- mod-version:3
+local syntax = require "core.syntax"
+
+syntax.add {
+  name = "Haxe Compiler Arguments",
+  files = "%.hxml$",
+  patterns = {
+    { pattern = "%-[%-%w_]*",      type="keyword"},
+    { pattern = "%.()%u[%w_]*",    type = {"normal", "keyword2"}},
+  },
+  symbols = {}
+}
+
+syntax.add {
+  name = "Haxe String Interpolation",
+  files = "%.hx__string_interp$",
+  patterns = {
+    { pattern = {"%${", "}", "\\"},     type="keyword", syntax = ".hx" },
+    { pattern = {"%$", "%s", "\\"},     type="keyword", syntax = ".hx" },
+    { pattern = "[^ ]",                 type = "string"}
+  },
+  symbols = {}
+}
+
+syntax.add {
+  name = "Haxe Regular Expressions",
+  files = "%.hx__regex$",
+  patterns = {
+    { pattern = "[%[%]%(%)]",           type = "string" },
+    { pattern = "[%.%*%+%?%^%$%|%-]",   type = "operator" },
+  },
+  symbols = {}
+}
+
+syntax.add {
+  name = "Haxe",
+  files = "%.hx$",
+  comment = "//",
+  patterns = {
+    { pattern = {"%~%/", "%/[igmsu]*"},                     type = "keyword2", syntax = ".hx__regex" },
+    { pattern = "%.%.%.",                                   type = "operator" },
+    { pattern = "%<()%u[%w_]*()%>*",                        type = {"operator", "keyword2", "operator"}},
+    { pattern = "%#%s*[%a_]*().*\n",                        type = {"keyword", "normal"}   },
+    { pattern = "import%s+()%u[%w]*",                       type = {"keyword2", "keyword2"}},
+    { pattern = "import%s+()[%w%.]*%.()%u[%w]*",            type = {"keyword2", "normal", "keyword2"}},
+    { pattern = "abstract%s+()%u[%w_]*%s*%(()%s*%u[%w_]*",  type = {"keyword2", "normal", "keyword2"} },
+    { pattern = "from%s+()%u[%w_]*%s+()to%s+()%u[%w_]*",    type = {"keyword", "keyword2", "keyword", "keyword2"}},
+    { pattern = "//.*\n",                                   type = "comment"  },
+    { pattern = { "/%*", "%*/" },                           type = "comment"  },
+    { pattern = { '"', '"', '\\' },                         type = "string"   },
+    { pattern = { "'", "'", "\\" },                         type = "string", syntax = ".hx__string_interp"},  
+    { pattern = "-?%.?%d+",                                 type = "number"   },
+    { pattern = "-?0x%x+",                                  type = "number"   },
+    { pattern = "-?%d+%.[%deE]+",                           type = "number"   },
+    { pattern = "-?%d+[%deE]+",                             type = "number"   },
+    { pattern = "[%+%-%.=/%*%^%%<>!~|&]",                   type = "operator" },
+    { pattern = "[%a_][%w_]*%s*%f[(]",                      type = "function" },
+    { pattern = "[%a_][%w_]*",                              type = "symbol"   },
+    { pattern = ":()%u[%a_][%w_]*",                         type = {"normal", "keyword2"}},
+    { pattern = "@:[%a_][%w_]*%f[(]",                       type = "keyword"   },
+    { pattern = "%$type",                                   type = "keyword"   },
+  },
+  symbols = {
```

Comment:
I agree. I picked keyword2 because I was using a Sublime Text 3 window to compare while making it, but I just noticed now that the color scheme I typically use there doesn't distinguish between them for some reason, but the others do... so I changed it so `class` `import` and `package` use `keyword`